esp_hosted_ng/host/main.c

```diff
@@ -720,6 +720,7 @@ static void process_rx_packet(struct esp_adapter *adapter, struct sk_buff *skb)
 			eap_skb = alloc_skb(skb->len + ETH_HLEN, GFP_KERNEL);
 			if (!eap_skb) {
 				esp_info("%u memory alloc failed\n", __LINE__);
+				dev_kfree_skb_any(skb);
```

Comment:
if the alloc_skb was failed, the eap_skb is already NULL. Not sure why should it be freed?

Reply:
I don't free eap_skb, as you may see.